apps/ideas/views.py

```diff
@@ -203,7 +203,63 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class IdeaSketchListView(generic.ListView):
-    queryset = IdeaSketch.objects.annotate_comment_count()
+class IdeaMixin(generic.detail.SingleObjectMixin):
+    model = Idea
+
+    def dispatch(self, request, *args, **kwargs):
+        self.idea = self.get_object()
+        self.object = self.idea
+        return super().dispatch(request, *args, **kwargs)
+
+
+class ProposalCreateWizard(PermissionRequiredMixin,
+                           SessionWizardView,
+                           IdeaMixin):
+    permission_required = 'advocate_europe_ideas.add_ideasketch'
+    file_storage = FileSystemStorage(
+        location=os.path.join(settings.MEDIA_ROOT, 'idea_sketch_images'))
+    title = _('create a proposal')
+
+    def get_form_initial(self, step):
+        initial = self.initial_dict.get(step, {})
+        form = self.form_list[step]
+        for field in form.base_fields:
+            if hasattr(self.idea, field):
+                initial[field] = getattr(self.idea, field)
+        return initial
+
+    def done(self, form_list, **kwargs):
```

Comment:
I think the method could be broken apart into more methods, or parts of it even be moved to quersets or models (eg. like `IdeaSketch.archive()` or `Proposals.objects.create_from_ideas(idea)`. In my opinion the archiving should also be able to be called twice, just in case that this method crashes between archiving and creating an proposal.